pkg/metrics/real_time_test.go

```diff
@@ -380,3 +382,182 @@ type mockRoundTripper struct {
+			InterfaceStats: stats.InterfaceStats{
+				Name:     "eth0",
+				RxBytes:  newUInt64Pointer(1024),
+				RxErrors: newUInt64Pointer(0),
+				TxBytes:  newUInt64Pointer(2048),
+				TxErrors: newUInt64Pointer(0),
+			},
+			Interfaces: []stats.InterfaceStats{
+				{
+					Name:     "eth1",
+					RxBytes:  newUInt64Pointer(64),
+					RxErrors: newUInt64Pointer(0),
+					TxBytes:  newUInt64Pointer(128),
+					TxErrors: newUInt64Pointer(0),
+				},
+				{
+					Name:     "wifi",
+					RxBytes:  newUInt64Pointer(2048),
+					RxErrors: newUInt64Pointer(0),
+					TxBytes:  newUInt64Pointer(4096),
+					TxErrors: newUInt64Pointer(0),
+				},
+			},
+		},
+	}
+
+	testDescription := "Successfully converts Extension Pod Stats to Kubelet Pod Stats"
+	t.Run(testDescription, func(t *testing.T) {
+		kubeletPodStats := extensionPodStatsToKubeletPodStats(pod, podStats)
+		assert.DeepEqual(t, expectedKubeletPodStats.PodRef.Name, kubeletPodStats.PodRef.Name)
+		assert.DeepEqual(t, expectedKubeletPodStats.StartTime, kubeletPodStats.StartTime)
+		assert.DeepEqual(t, expectedKubeletPodStats.CPU.Time, kubeletPodStats.CPU.Time)
+		assert.DeepEqual(t, expectedKubeletPodStats.CPU.UsageCoreNanoSeconds, kubeletPodStats.CPU.UsageCoreNanoSeconds)
+		assert.DeepEqual(t, expectedKubeletPodStats.CPU.UsageNanoCores, kubeletPodStats.CPU.UsageNanoCores)
+		assert.DeepEqual(t, expectedKubeletPodStats.Memory.UsageBytes, kubeletPodStats.Memory.UsageBytes)
+		assert.DeepEqual(t, expectedKubeletPodStats.Network.InterfaceStats.Name, kubeletPodStats.Network.InterfaceStats.Name)
+		assert.DeepEqual(t, expectedKubeletPodStats.Network.InterfaceStats.TxBytes, kubeletPodStats.Network.InterfaceStats.TxBytes)
+		assert.DeepEqual(t, expectedKubeletPodStats.Network.Interfaces[0].Name, kubeletPodStats.Network.Interfaces[0].Name)
+		assert.DeepEqual(t, expectedKubeletPodStats.Network.Interfaces[0].TxBytes, kubeletPodStats.Network.Interfaces[0].TxBytes)
+		assert.DeepEqual(t, expectedKubeletPodStats.Network.Interfaces[1].Name, kubeletPodStats.Network.Interfaces[1].Name)
+		assert.DeepEqual(t, expectedKubeletPodStats.Network.Interfaces[1].TxBytes, kubeletPodStats.Network.Interfaces[1].TxBytes)
+		assert.DeepEqual(t, expectedKubeletPodStats.Containers[0].CPU.UsageCoreNanoSeconds, kubeletPodStats.Containers[0].CPU.UsageCoreNanoSeconds)
+		assert.DeepEqual(t, expectedKubeletPodStats.Containers[0].CPU.UsageNanoCores, kubeletPodStats.Containers[0].CPU.UsageNanoCores)
+		assert.DeepEqual(t, expectedKubeletPodStats.Containers[0].Memory.UsageBytes, kubeletPodStats.Containers[0].Memory.UsageBytes)
+		assert.DeepEqual(t, expectedKubeletPodStats.Containers[1].CPU.UsageCoreNanoSeconds, kubeletPodStats.Containers[1].CPU.UsageCoreNanoSeconds)
+		assert.DeepEqual(t, expectedKubeletPodStats.Containers[1].CPU.UsageNanoCores, kubeletPodStats.Containers[1].CPU.UsageNanoCores)
+		assert.DeepEqual(t, expectedKubeletPodStats.Containers[1].Memory.UsageBytes, kubeletPodStats.Containers[1].Memory.UsageBytes)
```

Comment:
Is it possible to follow the same pattern we have [here](https://github.com/virtual-kubelet/azure-aci/pull/521/files#diff-3334415a6b3f7a408247d09c1a31e587ae6126cd38c5b74aa71b62498bc2f225R314)?